pkg/miniogw/gateway-storj.go

```diff
@@ -72,48 +204,104 @@ func (s *storjObjects) GetObject(ctx context.Context, bucket, object string,
-	object string) (objInfo cmd.ObjectInfo, err error) {
+	object string) (objInfo minio.ObjectInfo, err error) {
 	panic("TODO")
 }
 
 func (s *storjObjects) ListBuckets(ctx context.Context) (
-	buckets []cmd.BucketInfo, err error) {
-	return []cmd.BucketInfo{{
-		Name:    "test-bucket",
-		Created: time.Now(),
-	}}, nil
+	buckets []minio.BucketInfo, err error) {
+	return s.getBuckets()
 }
 
 func (s *storjObjects) ListObjects(ctx context.Context, bucket, prefix, marker,
-	delimiter string, maxKeys int) (result cmd.ListObjectsInfo, err error) {
-	return cmd.ListObjectsInfo{
-		IsTruncated: false,
-		Objects: []cmd.ObjectInfo{{
-			Bucket:      "test-bucket",
-			Name:        "test-file",
-			ModTime:     time.Now(),
-			Size:        0,
-			IsDir:       false,
-			ContentType: "application/octet-stream",
-		}},
-	}, nil
+	delimiter string, maxKeys int) (result minio.ListObjectsInfo, err error) {
+	return s.getFiles(bucket)
 }
 
 func (s *storjObjects) MakeBucketWithLocation(ctx context.Context,
 	bucket string, location string) error {
 	panic("TODO")
 }
 
+//encryptFile encrypts the uploaded files
+func encryptFile(data io.ReadCloser, blockSize uint, bucket, object string) error {
+	dir := os.TempDir()
+	dir = dir + "gateway/" + bucket + "/" + object
+	fmt.Println("temp dir = ", dir)
+	err := os.MkdirAll(dir, 0755)
+	if err != nil {
+		return err
+	}
+	fc, err := infectious.NewFEC(*rsk, *rsn)
+	if err != nil {
+		return err
+	}
+	es := eestream.NewRSScheme(fc, *pieceBlockSize)
+	encKey := sha256.Sum256([]byte(*key))
+	var firstNonce [12]byte
+	encrypter, err := eestream.NewAESGCMEncrypter(
+		&encKey, &firstNonce, es.DecodedBlockSize())
+	if err != nil {
+		return err
+	}
+	readers := eestream.EncodeReader(eestream.TransformReader(
+		eestream.PadReader(data, encrypter.InBlockSize()), encrypter, 0), es)
+	errs := make(chan error, len(readers))
+	for i := range readers {
+		go func(i int) {
+			fh, err := os.Create(
+				filepath.Join(dir, fmt.Sprintf("%s%d.piece", object, i)))
+			if err != nil {
+				errs <- err
+				return
+			}
+			defer fh.Close()
+			_, err = io.Copy(fh, readers[i])
+			errs <- err
+		}(i)
+	}
+	for range readers {
+		err := <-errs
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func (s *storjObjects) PutObject(ctx context.Context, bucket, object string,
-	data *hash.Reader, metadata map[string]string) (objInfo cmd.ObjectInfo,
+	data *hash.Reader, metadata map[string]string) (objInfo minio.ObjectInfo,
 	err error) {
-	panic("TODO")
+	srcFile := path.Join(s.TempDir, minio.MustGetUUID())
+	writer, err := os.Create(srcFile)
+	if err != nil {
+		return objInfo, err
+	}
+
+	wsize, err := io.CopyN(writer, data, data.Size())
+	if err != nil {
+		os.Remove(srcFile)
+		return objInfo, err
+	}
+
+	err = encryptFile(writer, uint(wsize), bucket, object)
+	if err == nil {
+		// return values are for debugging purposes
+		_, _ = s.uploadFile(bucket, object, wsize, metadata)
```

Comment:
I don't see these return values used for anything. Perhaps, remove them together with the comment.

Reply:
done